gateway/gateway.go

```diff
@@ -23,10 +21,6 @@ type ServerData struct {
 	ConnCh chan<- connection.HandshakeConn
 }
 
-type Gateway interface {
-	HandleConn(conn connection.HandshakeConn) error
-}
```

Comment:
Not sure why you removed it, but having an interface like this could be really powerful.

Reply:
I asked myself: "Is this really necessary to have right now? Do we have multiple structs which are using this interface right now?" Im sure the answer on those questions are "no" so i deleted it. We can always add it back when we need it.